src/gems/simulation/output_values.py

```diff
@@ -1,155 +1,82 @@
+        key = TimeScenarioIndex(timestep, scenario)
+        if key not in self._value:
+            size_s = max(self._size[0], scenario + 1)
+            size_t = max(self._size[1], timestep + 1)
+            self._size = (size_s, size_t)
 
-        @property
-        def value(self) -> Union[None, float, List[float], List[List[float]]]:
-            size_s, size_t = self._size
-            if size_t == 1:
-                if size_s == 1:
-                    # Constant
-                    return self._value[TimeScenarioIndex(0, 0)]
-                else:
-                    # Scenario-only
-                    return [self._value[TimeScenarioIndex(0, s)] for s in range(size_s)]
-            else:
-                # Either Time-only or Time-Scenario
-                return [
-                    [self._value[TimeScenarioIndex(t, s)] for t in range(size_t)]
-                    for s in range(size_s)
-                ]
-
-        @value.setter
-        def value(self, values: Union[float, List[float], List[List[float]]]) -> None:
-            size_s, size_t = 1, 1
-
-            if isinstance(values, list):
-                size_s = len(values)
-                for scenario, timesteps in enumerate(values):
-                    if isinstance(timesteps, list):
-                        size_t = len(timesteps)
-                        for timestep, value in enumerate(timesteps):
-                            # Either Time-only or Time-Scenario
-                            self._value[TimeScenarioIndex(timestep, scenario)] = value
-                    else:
-                        # Scenario-only
-                        self._value[TimeScenarioIndex(0, scenario)] = cast(
-                            float, timesteps
-                        )
-            else:
-                # Constant
-                self._value[TimeScenarioIndex(0, 0)] = values
+        self._value[key] = value
+        if not is_mip and status is not None:
+            self._basis_status[key] = status
 
-            self._size = (size_s, size_t)
 
-        def _set(
-            self,
-            timestep: Optional[int],
-            scenario: Optional[int],
-            value: float,
-            status: Optional[str] = None,
-            is_mip: bool = True,
-        ) -> None:
-            timestep = 0 if timestep is None else timestep
-            scenario = 0 if scenario is None else scenario
-            key = TimeScenarioIndex(timestep, scenario)
-            if key not in self._value:
-                size_s = max(self._size[0], scenario + 1)
-                size_t = max(self._size[1], timestep + 1)
-                self._size = (size_s, size_t)
-
-            self._value[key] = value
-            if not is_mip and status is not None:
-                self._basis_status[key] = status
+@dataclass
+class OutputValues:
+    """
+    Contains variables and extra outputs after solver work completion.
+    """
 
     @dataclass
     class Component:
```

Comment:
Try to be consistent in your design choicies : either define Variable, ExtraOutput and Component all outside the OutputValues class or all inside, but not half/half. And the project has already other classes named Variable, and Component, so be sure there are no naming conflicts if classes are defined outside